dashboard/app/models/concerns/text_to_speech.rb

```diff
@@ -221,6 +221,7 @@ def tts_authored_hints_texts
   end
 
   def tts_update(update_all = false)
+    return unless Rails.application.config.levelbuilder_mode
```

Comment:
I'm guessing you're doing this to prevent

```
  parent_levels.contained.each do |containing_level|
    containing_level.tts_upload_to_s3(containing_level.tts_long_instructions_text, context)
  end
```
from running. Does it cause problems if it runs, or just slows things down?
from running.

Reply:
correct! all of the above :-) TTS is expensive, so we wouldn't want to accidentally generate speech twice. also there is the chance for another environment to overwrite newer content with older content. I wasn't able to prove or disprove that this would actually happen if someone added the CDO keys to another environment, so it seemed better to add a safeguard while I was looking at this rather than leaving a sharp corner for someone else to stumble into.

Reply:
Ah, thanks for the context

Reply:
I'm not entirely convinced we want to do this. We do [check to see if we've already generated speech for a given filepath](https://github.com/code-dot-org/code-dot-org/blob/5ce06dbb799ba65ef633e0128caff26d7e5843dc/dashboard/app/models/concerns/text_to_speech.rb#L94) before invoking the expensive bit, and there's no chance of content being overwritten because we [use a hash of the content as part of the filepath](https://github.com/code-dot-org/code-dot-org/blob/5ce06dbb799ba65ef633e0128caff26d7e5843dc/dashboard/app/models/concerns/text_to_speech.rb#L137-L139).

I'd also much rather not introduce more hardcoded references to specific environments into our codebase if we can avoid it.

Reply:
thanks for digging into that, Elijah! That addresses both of my concerns, and I've reverted that part of the PR.